spring-cloud-alibaba-sentinel-zuul/src/main/java/org/springframework/cloud/alibaba/sentinel/zuul/SentinelZuulAutoConfiguration.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.alibaba.sentinel.zuul;
+
+import com.alibaba.csp.sentinel.adapter.zuul.fallback.DefaultRequestOriginParser;
+import com.alibaba.csp.sentinel.adapter.zuul.fallback.DefaultUrlCleaner;
+import com.alibaba.csp.sentinel.adapter.zuul.fallback.RequestOriginParser;
+import com.alibaba.csp.sentinel.adapter.zuul.fallback.UrlCleaner;
+import com.alibaba.csp.sentinel.adapter.zuul.filters.SentinelErrorFilter;
+import com.alibaba.csp.sentinel.adapter.zuul.filters.SentinelPostFilter;
+import com.alibaba.csp.sentinel.adapter.zuul.filters.SentinelPreFilter;
+import com.alibaba.csp.sentinel.adapter.zuul.properties.SentinelZuulProperties;
+import com.alibaba.csp.sentinel.util.StringUtil;
+import com.netflix.zuul.ZuulFilter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.alibaba.sentinel.zuul.listener.FallBackProviderListener;
+import org.springframework.cloud.netflix.zuul.filters.ProxyRequestHelper;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+import static org.springframework.cloud.alibaba.sentinel.zuul.SentinelZuulAutoConfiguration.PREFIX;
+
+/**
+ * Sentinel Spring Cloud Zuul AutoConfiguration
+ *
+ * @author tiger
+ */
+@Configuration
+@ConditionalOnProperty(prefix = PREFIX, name = "enabled", havingValue = "true")
+public class SentinelZuulAutoConfiguration {
+
+    @Autowired
+    private Environment environment;
+
+    public static final String PREFIX = "spring.cloud.alibaba.sentinel.zuul";
+
+    @Bean
+    public SentinelZuulProperties sentinelZuulProperties() {
```

Comment:
Why don't use @ConfigurationProperties and @EnableConfigurationProperties?

Reply:
SentinelZuulProperties is not annotated by @ConfigurationProperties

Reply:
we use sentinel-zuul-adapter as a dependency which can be used in independent. and it`s not a spring project.